### PR TITLE
Move refresh just before _check (fixes #124 for multi level BoM as well)

### LINKED ISSUE #124: is:issue is:open [8.0][mrp_production_real_cost] Error when produce product
Hi,

there's new error from mrp_production_real_cost, after I do git pull from the last commit 

```

ERROR demo1 openerp.sql_db: Programming error: can't adapt type 'mrp.production', in query  SELECT "mrp_production"."id" FROM "mrp_production"
                        WHERE "mrp_production".id IN %s  ORDER BY "mrp_production"."priority" DESC,"mrp_production"."date_planned" ASC         

  File "/opt/odoo/server/addons/mrp_production_real_cost/models/mrp_production.py", line 34, in action_production_end
    self.mapped('move_created_ids2').filtered(


  File "/usr/lib/python2.7/dist-packages/psycopg2/extensions.py", line 129, in getquoted
    pobjs = [adapt(o) for o in self._seq]
ValueError: "can't adapt type 'mrp.production'" while evaluating
u'action_production_end()'
```

regards

### PR DESCRIPTION
**Pull request by levkar**
fixes https://github.com/OCA/manufacture/issues/124 for multi level BoM's.

### CODE REVIEW

File: mrp_production_real_cost/models/procurement_order.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    def _check(self, cr, uid, procurement, context=None):
+        if procurement.production_id:
+            # This is needed because commit
+            # https://github.com/odoo/odoo/commit/
+            # 6f29bfc181d23d70d29776d96b4318e9ee2c93a9
+            # introduces a weird behavior on the next call, provoking an error
+            procurement.production_id.sudo().refresh()
+        return super(ProcurementOrder, self)._check(
+            cr, uid, procurement, context=context)
```
Comment (pedrobaeza) [inline mrp_production_real_cost/models/procurement_order.py]: Line at the end

File: mrp_production_real_cost/models/procurement_order.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# © 2016 Eska Yazılım ve Danışmanlık A.Ş. - Levent Karakaş
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models
+
+
+class ProcurementOrder(models.Model):
+    _inherit = 'procurement.order'
+
+    def _check(self, cr, uid, procurement, context=None):
```
Comment (pedrobaeza) [inline mrp_production_real_cost/models/procurement_order.py]: Why not use @api.model and remove old API?

**Comment by pedrobaeza** [comment]
So it seems that you need the refresh on both places?

Please squash all the commits in one.

**Comment by oihane** [comment]
LGTM :+1:

**Comment by levkar** [comment]
Figuring out squashing took more time than the fix :)

**Comment by pedrobaeza** [comment]
Is this still needed with recent revisions?

**Comment by levkar** [comment]
I haven't tested with final version but I cannot see any PR which fixes this problem and later than this PR.

Seems like nobody is using multi level BoM :)

**Comment by pedrobaeza** [comment]
@oihane @anajuaristi maybe you can check?

**Comment by LoisRForgeFlow** [comment]
ping @oihane @anajuaristi

**Comment by LoisRForgeFlow** [comment]
Closing as there are unresolved comments for a long time.

**Pull request closed without merging**